heartbeat/monitors/active/http/respbody.go

```diff
@@ -94,28 +95,22 @@ func readBody(resp *http.Response, maxSampleBytes int) (bodySample string, bodyS
 
 func readPrefixAndHash(body io.ReadCloser, maxPrefixSize int) (respSize int, prefix string, hashStr string, err error) {
 	hash := sha256.New()
-	// Function to lazily get the body of the response
-	rawBuf := make([]byte, 1024)
 
-	// Buffer to hold the prefix output along with tracking info
-	prefixBuf := make([]byte, maxPrefixSize)
-	prefixRemainingBytes := maxPrefixSize
-	prefixWriteOffset := 0
+	var prefixBuf bytes.Buffer
```

Comment:
tip: consider `strings.Builder`. `strings.Builder` also implements an `io.Reader`. But when calling `(bytes.Buffer).String()` you create a copy, while `(strings.Builder).String()` returns you the buffer as is (no extra copy).

Reply:
Thanks! Switched to a `strings.Builder`